pkg/util/debouncer/debouncer_test.go

```diff
@@ -0,0 +1,214 @@
+package debouncer
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDebouncer(t *testing.T) {
+	t.Run("should process values after min wait", func(t *testing.T) {
+		var processedMu sync.Mutex
+		processedValues := make(map[string]int)
+
+		group, err := NewGroup(DebouncerOpts[string]{
+			BufferSize: 10,
+			KeyFunc:    func(s string) string { return s },
+			ProcessHandler: func(ctx context.Context, value string) error {
+				processedMu.Lock()
+				processedValues[value]++
+				processedMu.Unlock()
+				return nil
+			},
+			MinWait: 10 * time.Millisecond,
+			MaxWait: 100 * time.Millisecond,
+		})
+		require.NoError(t, err)
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		group.Start(ctx)
+
+		require.NoError(t, group.Add("key1"))
+		require.NoError(t, group.Add("key2"))
+		// Should be deduplicated.
+		require.NoError(t, group.Add("key1"))
+
+		// Wait for processing.
+		time.Sleep(20 * time.Millisecond)
+
+		// We should have processed key1 and key2 exactly once.
+		processedMu.Lock()
+		require.Equal(t, 1, processedValues["key1"])
+		require.Equal(t, 1, processedValues["key2"])
+		processedMu.Unlock()
+	})
+
+	t.Run("should process values after max wait", func(t *testing.T) {
+		processed := make(map[string]int, 1)
+
+		group, err := NewGroup(DebouncerOpts[string]{
+			BufferSize: 10,
+			KeyFunc:    func(s string) string { return s },
+			ProcessHandler: func(ctx context.Context, value string) error {
+				processed[value]++
+				return nil
+			},
+			MinWait: 100 * time.Millisecond,
+			// Test max wait by setting it lower than min wait.
```

Comment:
That's interesting test... shouldn't we enfore MinWait <= MaxWait?

Reply:
I changed it 👍